src/data/index.ts

```diff
@@ -96,7 +96,7 @@ export function getCollaborationLinkData(link: string) {
 export async function generateCollaborationLink() {
   const id = await generateRandomID();
   const key = await generateEncryptionKey();
-  return `${window.location.origin}#room=${id},${key}`;
+  return `${window.location.protocol}//${window.location.host}${window.location.pathname}#room=${id},${key}`;
```

Comment:
Can you keep `location.origin` and simply append the `pathname`?

Reply:
@dwelle 

Why? Would there be side effects or for compatibility reasons?

Reply:
In my opinion it is a safer way to build a path.

Reply:
Likely not, but it's simpler. Was there a reason you expanded it to protocol & host?

Reply:
> In my opinion it is a safer way to build a path.

Quite the opposite. It's safer to delegate it to something not built by *us*.

Reply:
Right, changed it. Thanks 👍